cdap-master/src/main/java/co/cask/cdap/data/tools/UpgradeTool.java

```diff
@@ -417,9 +424,9 @@ private void initializeDSFramework(CConfiguration cConf,
                                      boolean includeNewDatasets) throws IOException, DatasetManagementException {
     // dataset service
     DatasetMetaTableUtil.setupDatasets(datasetFramework);
+    ArtifactStore.setupDatasets(datasetFramework);
```

Comment:
@anew : I moved the ArtifactStore.setupDatasets(datasetFramework); out of the if condition as in 3.2 this table was created and 3.3 it will be there.